System.Compression.Primitives/System/IO/Compression/Brotli/BrotliStream.cs

```diff
@@ -138,8 +111,7 @@ protected virtual void FlushEncoder(Boolean finished)
             UInt32 totalOut = 0;
             while (true)
             {
-                var compressOK = BrotliNative.BrotliEncoderCompressStream(_encoder.State, op, ref AvailIn, ref NextIn, ref AvailOut, ref NextOut, out totalOut);
-                if (!compressOK) throw new Exception();// unable encode
+                if (!BrotliNative.BrotliEncoderCompressStream(_encoder.State, op, ref AvailIn, ref NextIn, ref AvailOut, ref NextOut, out totalOut)) throw new Exception();// unable encode
```

Comment:
never throw plain Exception - please make this an IOException with an error message.  See https://msdn.microsoft.com/en-us/library/hh875103(v=vs.110).aspx for an example of what other IO.Compression APIs can throw in error situations.